ksqldb-engine/src/main/java/io/confluent/ksql/security/KsqlAuthorizationValidatorImpl.java

```diff
@@ -69,10 +75,9 @@ private void validateQuery(
       final MetaStore metaStore,
       final Query query
   ) {
-    final SourceTopicsExtractor extractor = new SourceTopicsExtractor(metaStore);
-    extractor.process(query, null);
-    for (String kafkaTopic : extractor.getSourceTopics()) {
-      accessValidator.checkAccess(securityContext, kafkaTopic, AclOperation.READ);
+    for (KsqlTopic ksqlTopic : extractQueryTopics(query, metaStore)) {
```

Comment:
This is not a comment but a question for my own understanding: it seems `KsqlAuthorizationValidator` is wrapping a `KsqlAccessValidator` here, can we just merge these two interfaces? I guess that's boiled down to:

* What are the conceptual differences of these two?
* Would `KsqlAccessValidator` ever need to be leveraged directly without the `KsqlAuthorizationValidator`?

Reply:
`KsqlAuthorizationValidator` validates the user can execute a specified statement. This class gets information about that, such as topics and schemas; and then validates each resource based on the type of statement (query, create, insert, ...).

Initially, the validation was part of this `KsqlAuthorizationValidator`, but I split it when I introduced an authorization cache, which lives in `KsqlCacheAccessValidator`. The cache wraps another validation if the cache is enabled. The current validation, besides caching, is the `KsqlBackendAccessValidator`, which just checks with the backend service (i.e. Kafka) to check for ACLs. The new one is `KsqlProvidedAccessValidator`, which checks with an external service, like RBAC. SR permissions will be checked by the KSQL confluent security plugins which can either use RBAC or call the SR `/permissions` endpoint to check for ACLs (both part of the confluent security plugins).